# HG changeset patch
# User Jacob Dawid <jacob.dawid@gmail.com>
# Date 1342645619 14400
#      Wed Jul 18 17:06:59 2012 -0400
# Branch gui
# Node ID bc6c099eacc0321f5ae56c9b0ed6651543379e76
# Parent  3fd857c284fe9571035554386aa73fb2267396b9
Removed cd'ing back after running a script for compatibility with less.

* file-editor-tab.cc: Removed cd'ing back after running the m-file.

diff --git a/gui/src/m-editor/file-editor-tab.cc b/gui/src/m-editor/file-editor-tab.cc
--- a/gui/src/m-editor/file-editor-tab.cc
+++ b/gui/src/m-editor/file-editor-tab.cc
@@ -720,17 +720,17 @@ file_editor_tab::run_file ()
   QFileInfo file_info (_file_name);
   QString path = file_info.absolutePath ();
   QString current_path = QString::fromStdString
       (octave_link::instance ()->get_last_working_directory ());
   QString function_name = file_info.fileName ();
 
   // We have to cut off the suffix, because octave appends it.
   function_name.chop (file_info.suffix ().length () + 1);
-  _file_editor->terminal ()->sendText (QString ("cd \'%1\'\n%2\ncd \'%3\'\n\n")
+  _file_editor->terminal ()->sendText (QString ("cd \'%1\'\n%2\n")
     .arg(path).arg (function_name).arg (current_path));
   // TODO: Sending a run event crashes for long scripts. Find out why.
   //  octave_link::instance ()
   //      ->post_event (new octave_run_file_event (*this, _file_name.toStdString ()));
 }
 
 void
 file_editor_tab::file_has_changed (QString fileName)
